sds_data_manager/stacks/indexer_lambda_stack.py

```diff
@@ -76,3 +80,92 @@ def __init__(
             self, "rds_secret", db_secret_name
         )
         rds_secret.grant_read(grantee=indexer_lambda)
+
+        # Instrument list
+        instruments = [
+            "codice",
+            "glows",
+            "hi",
+            "hit",
+            "idex",
+            "lo",
+            "mag",
+            "swapi",
+            "swe",
+            "ultra",
+        ]
+
+        # Events that triggers Indexer Lambda:
+        # 1. Arrival of L0 data
+        # 2. PutEvent from Lambda that builds dependency and starts Batch Job
+        # 3. Batch Job status change
+
+        # Write l0 info to db with
+        # status SUCCEEDED
+        l0_arrival_rule = events.Rule(
+            self,
+            "l0DataArrival",
+            rule_name="l0-data-arrival",
+            event_pattern=events.EventPattern(
+                source=["aws.s3"],
+                detail_type=["Object Created"],
+                detail={
+                    "bucket": {"name": [data_bucket.bucket_name]},
+                    "object": {
+                        "key": [
+                            {"prefix": f"imap/{instrument}/l0/"}
+                            for instrument in instruments
+                        ]
+                    },
+                },
+            ),
+        )
+
+        # This event listens for PutEvent from Lambda that builds
+        # dependency and starts Batch Job.
+        # Indexer Lambda listens for that event and writes
+        # all information to database
+        # and intialized other information such as
+        # ingestion time to null.
+        # PutEvent Example:
+        #     {
+        #     "DetailType": "Job Started",
+        #     "Source": "aws.lambda",
+        #     "Detail": {
+        #     "filename": "str",
+        #     "status": "InProgress",
+        #     "depedency": {    "codice": "s3-filepath", "mag": "s3-filepath"}
+        #     }}
+        batch_starter_event_rule = events.Rule(
+            self,
+            "batchStarterEvent",
+            rule_name="batch-starter-event",
+            event_pattern=events.EventPattern(
+                source=["imap.lambda"],
+                detail_type=["Batch Job Started"],
+                detail={
+                    "filename": [{"exists": True}],
+                    "status": ["InProgress"],
+                    "depedency": [{"exists": True}],
```

Comment:
Put something as a placeholder for now?

Also, `filename` is a bit unclear to me which file we are talking about. I think it is the file you're creating? Could you add some naming to make that clear in the variable we are passing around?

```suggestion
                    "file_to_create": "imap_mag_l1a_...",
                    "status": ["InProgress"],
                    "depedency": [{"mag": "imap_mag_l0_raw_..."}],
```

Now that I'm thinking looking at this more closely, is it actually saying only trigger the event if the detail has the fields "filename", "status", and "dependency" (the `"exists": True`)? If so, maybe we don't need all of these for the event rule, but just need the detail_type for the match. Ignore my suggestion if this is the case.

Reply:
Yes, latter. It only triggers target only when those field exits. We don't do checks there. We will do the checks in indexer lambda or any target we specify. But I do agree with better naming. I get confused too. I like `file_to_create` name. I will change that.